dataflux_pytorch/benchmark/checkpointing/simulated/llama2.py

```diff
@@ -0,0 +1,119 @@
+
+    configure_master_addr()
+    # Using gloo backend since the simulated version runs on CPU.
+    torch.distributed.init_process_group(backend='gloo',
+                                         rank=rank,
+                                         world_size=world_size)
+
+
+def run_benchmark(world_size: int, project: str, filepath: str,
+                  sample_count: int, use_fsspec: bool,
+                  model_parameter_size: str, optimizer: str) -> None:
+
+    if os.environ.get("COORDINATOR_ADDRESS"):
+        init_processes()
+    rank = int(os.environ.get("NODE_RANK", 0))
+
+    benchmark_strategy = BenchmarkStrategy(project=project,
+                                           path=filepath,
+                                           use_fsspec=use_fsspec)
+    print(f'Constructing state dict for LLAMA2 {model_parameter_size}')
+    state_dict = create_llama2_state_dict(world_size=world_size,
+                                          rank=rank,
+                                          parameters=model_parameter_size,
+                                          optimizer=optimizer,
+                                          empty=False)
+
+    dist.barrier()
+    save_checkpoint_times = time_checkpoint_operation(
+        benchmark_strategy, state_dict, filepath, sample_count, 'save', rank,
+        world_size, model_parameter_size, optimizer)
+
+    load_checkpoint_times = time_checkpoint_operation(
+        benchmark_strategy, state_dict, filepath, sample_count, 'load', rank,
+        world_size, model_parameter_size, optimizer)
+
+    if rank == 0:
+        print(f"Time taken to save checkpoint:\
+                {statistics.mean(save_checkpoint_times):.4f} seconds (stdev {statistics.stdev(save_checkpoint_times):.4f})"
+              )
+        print(f"All save times: {save_checkpoint_times}")
+        print(f"Time taken to load checkpoint:\
+                 {statistics.mean(load_checkpoint_times):.4f} seconds (stdev {statistics.stdev(load_checkpoint_times):.4f})"
+              )
+        print(f"All load times: {load_checkpoint_times}")
+
+        tensor_size_per_instance = sum(v.element_size() * v.numel()
+                                       for v in state_dict.values())
+        total_size_bytes = tensor_size_per_instance * world_size
+
+        print(f"Size of distributed tensors (rank {rank}):\
+                 {format_size(tensor_size_per_instance)}")
+        print(f"Total size of all tensors:\
+                 {format_size(total_size_bytes)}")
+
+        print("######################")
+    cleanup()
+
+
+def main() -> None:
+    world_size = int(os.getenv("WORLD_SIZE"))
+    project = os.getenv("PROJECT")
+    ckpt_dir_path = os.getenv("CKPT_DIR_PATH")
+    sample_count = int(os.getenv("SAMPLE_COUNT", 8))
+    use_fsspec = os.getenv("USE_FSSPEC",
+                           "False").lower() in ("true", "1", "yes")
+    model_parameter_size = os.getenv("MODEL_PARAMATER_SIZE")
```

Comment:
nit: Consider making the model parameter sizes a tuple defined in llama2_utils which can be used for validation here and also with args.choices for the demo. This will help validate input sooner and help with documentation.

Reply:
For now, I think I want to keep all the variables limited to single file, instead of splitting them across llama2 and llama2_utils